homeassistant/components/sensor/transmission.py

```diff
@@ -116,6 +122,50 @@ def update(self):
         self._transmission_api.update()
         self._data = self._transmission_api.data
 
+        if self.first_run:
+            actual_torrents = self._transmission_api.torrents
+            actual_completed_torrents = [
+                x.name for x in actual_torrents if x.status == "seeding"]
+            actual_started_torrents = [
+                x.name for x in actual_torrents if x.status == "downloading"]
+            self.completed_torrents = actual_completed_torrents
+            self.started_torrents = actual_started_torrents
+
+        elif self.type == 'completed_torrents':
+            actual_torrents = self._transmission_api.torrents
+            actual_completed_torrents = [
+                var.name for var in actual_torrents if var.status == "seeding"]
+
+            tmp_completed_torrents = list(
+                set(actual_completed_torrents).difference(
+                    self.completed_torrents))
+            if tmp_completed_torrents:
+                for var in tmp_completed_torrents:
+                    self.hass.bus.fire(
```

Comment:
Do not fire events inside the update method. If you want to fire events, you need to implement a component.

Reply:
I tried to implement what @kellerza suggested, so I move the torrents lists and the fire of the event inside the TransmissionData class, so its called and instantiated less times. Does it look ok for you? Otherwhise, what do you mean by “if you want to fire event, you need to implement a component”?

Reply:
`homeassistant/components/transmission.py`

Reply:
Since it's needed to implement a component in order to fire event, I think that this PR can be closed and I will try to create a new component for this feature.